ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/assertion/DirectionsQuestionnaireAssert.java

```diff
@@ -0,0 +1,160 @@
+package uk.gov.hmcts.cmc.ccd.assertion;
+
+import org.assertj.core.api.AbstractAssert;
+import uk.gov.hmcts.cmc.ccd.domain.directionsquestionnaire.CCDDirectionsQuestionnaire;
+import uk.gov.hmcts.cmc.domain.models.directionsquestionnaire.DirectionsQuestionnaire;
+import uk.gov.hmcts.cmc.domain.models.directionsquestionnaire.ExpertRequest;
+import uk.gov.hmcts.cmc.domain.models.directionsquestionnaire.HearingLocation;
+import uk.gov.hmcts.cmc.domain.models.directionsquestionnaire.RequireSupport;
+import uk.gov.hmcts.cmc.domain.models.directionsquestionnaire.Witness;
+
+import java.util.Objects;
+import java.util.function.Consumer;
+
+import static uk.gov.hmcts.cmc.ccd.assertion.Assertions.assertThat;
+
+public class DirectionsQuestionnaireAssert
+    extends AbstractAssert<DirectionsQuestionnaireAssert, DirectionsQuestionnaire> {
+
+    public DirectionsQuestionnaireAssert(DirectionsQuestionnaire actual) {
+        super(actual, DirectionsQuestionnaireAssert.class);
+    }
+
+    public DirectionsQuestionnaireAssert isEqualTo(CCDDirectionsQuestionnaire ccdDirectionsQuestionnaire) {
+
+        isNotNull();
+
+        isEqualToHearingLocation(ccdDirectionsQuestionnaire, actual.getHearingLocation());
+
+        actual.getRequireSupport().ifPresent(isEqualToRequireSupport(ccdDirectionsQuestionnaire));
+
+        actual.getWitness().ifPresent(isEqualToWitness(ccdDirectionsQuestionnaire));
+
+        actual.getExpertRequest().ifPresent(isEqualToExpertRequest(ccdDirectionsQuestionnaire));
+
+        return this;
+    }
+
+    private Consumer<Witness> isEqualToWitness(CCDDirectionsQuestionnaire ccdDirectionsQuestionnaire) {
+        return witness -> {
+            if (!Objects.equals(witness.getSelfWitness().name(), ccdDirectionsQuestionnaire.getSelfWitness().name())) {
+                failWithMessage("Expected DirectionsQuestionnaire.selfWitness to be <%s> but was <%s>",
+                    ccdDirectionsQuestionnaire.getSelfWitness(), witness.getSelfWitness());
+            }
+
+            witness.getNoOfOtherWitness().ifPresent(noOfOtherWitness -> {
+                    if (!Objects.equals(noOfOtherWitness, ccdDirectionsQuestionnaire.getHowManyOtherWitness())) {
+                        failWithMessage(
+                            "Expected DirectionsQuestionnaire.noOfOtherWitness to be <%s> but was <%s>",
+                            ccdDirectionsQuestionnaire.getHowManyOtherWitness(), witness.getNoOfOtherWitness());
+                    }
+                }
+            );
```

Comment:
When the entire body is wrapped in an `if` that's a good sign you could use a `filter` first

```
witness.getNoOfOtherWitness()
    .filter(noOfOtherWitness -> !Objects.equals(...))
    .ifPresent(noOfOtherWitness -> failWithMessage(...));
```